### PR TITLE
Add metrics to track connection ingress

### PR DESCRIPTION
**Pull request by dboslee**
Four metrics added
- accepted connections
- active connections
- authenticated accepted connections
- authenticated active connections

These track the service and path the connection came through. Differentiating between an alpn routed connection or one directly to the listener.

### Context
For cloud we currently have customers configured to use tls routing but have left the ports open for them to connect directly to the listeners for ssh/kube/postgres/mysql... We would like to remove direct access to these ports without impacting customers workflows. (Some customers may rely on these ports for openssh or some database GUI clients depending on how they have their clients configured). These metrics will help us identify any customers using these ports directly.

### CODE REVIEW

**Review by rosstimothy — COMMENTED** [review]
Did you consider implementing a `net.Listener` that automatically incremented `ConnectionAccepeted` and wrapped the `net.Conn` with one that decremented `ConnectionClosed` when it was closed? That might help reduce some of the surface area of these changes. I just did something similar for connection limiting in #20130

File: lib/srv/ingress/reporter.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+		return PathDirect
+	}
+	if r.alpnAddr == addr.String() {
+		return PathALPN
+	}
+	return PathDirect
+}
+
+// getRealLocalAddr returns the underlying local address if proxy protocol is being used.
+func getRealLocalAddr(conn net.Conn) net.Addr {
```
[the remark this answers is not in the document]
Comment (AntonAM) [reply lib/srv/ingress/reporter.go:192]: As I understand that one is about destination addresses, and in that RFD we mostly care about source address of a client. I'll probably add another layer to unwrap, but that's tangetial.

File: lib/srv/ingress/reporter.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+// Constants for each ingress service.
+const (
+	Web         = "web"
+	SSH         = "ssh"
+	Kube        = "kube"
+	Tunnel      = "tunnel"
+	MySQL       = "mysql"
+	Postgres    = "postgres"
+	SQLServer   = "sqlserver"
```
Comment (camscale) [inline lib/srv/ingress/reporter.go:39]: This is not used in the PR. Was recording metrics for this forgotten or is it to be done later?

**Review by camscale — APPROVED** [review]

File: lib/srv/ingress/reporter.go [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+		conn = tlsConn.NetConn()
+	}
+	// Uwrap a alpnproxy.bufferedConn without exporting or causing a circular dependency.
+	if connGetter, ok := conn.(netConnGetter); ok {
+		conn = connGetter.NetConn()
+	}
+	// Unwrap a multiplexer.Conn without exporting or causing a circular dependency..
+	if connGetter, ok := conn.(netConnGetter); ok {
+		conn = connGetter.NetConn()
+	}
```
[the remark this answers is not in the document]
Comment (AntonAM) [reply lib/srv/ingress/reporter.go:205]: I think loop and general pattern of putting `connGetter` to our custom conn types might be a good idea.

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**